packages/core/lib/src/models/tweet.dart

```diff
@@ -1,22 +1,33 @@
 import 'package:core/core.dart';
 import 'package:core/src/utils/twitter_regex.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'tweet.freezed.dart';
 
 /// Tweet object model
-class Tweet {
+@freezed
+class Tweet with _$Tweet {
   /// Creates a new instance of [Tweet]
-  const Tweet({
-    required this.rawText,
-    this.annotationType,
-    required this.author,
-    this.media = const [],
-    this.publicMetrics = const PublicMetrics(),
-    required this.createdAt,
-    this.entities = const TweetEntities(),
-    this.inReplyToUser,
-  });
+  const factory Tweet({
+    required String rawText,
+    TweetAnnotationType? annotationType,
+    required User author,
+    @Default([]) List<Media> media,
+    @Default(PublicMetrics()) PublicMetrics publicMetrics,
+    required DateTime createdAt,
+    @Default(TweetEntities()) TweetEntities entities,
+
+    /// Id of the user that this tweet is a reply to
+    ///
+    /// Only available when this tweet is a reply.
+    ///
+    /// If available, a `replying to @{inReplyToUser.username}` text
+    /// will be shown
+    User? inReplyToUser,
```

Comment:
this looks like it should be called `idReplyToUser`. I'd probably call this `replyToUserId`. but it's your choice

Reply:
The Twitter API returns the id in a key named `in_reply_to_user_id`, so I was trying to reflect it. I also removed the `id` from the name because the response also returns the user object and I was planning to return the whole `User` object directly instead of just the id value.